pages/upload.tsx

```diff
@@ -149,34 +148,38 @@ function Upload() {
   const onSubmit = async () => {
     setError(undefined)
 
-    const data = getStepsData(steps, true)
-    const form = new FormData()
+    if (steps.filter((e) => !e.isComplete(e)).length === 0) {
```

Comment:
I'd prefer if the layout was kept to the `if (error) { returnEarly }` approach, e.g:

```
if (!steps.every(e => e.isComplete(e)) {
  return setError("Ensure all steps are complete before submitting")
}

// otherwise, submit...
```